shared/data/src/main/java/com/ivy/data/repository/impl/TransactionRepositoryImpl.kt

```diff
@@ -861,4 +877,10 @@ class TransactionRepositoryImpl @Inject constructor(
         accountId ?: return null
         return accountRepository.findById(AccountId(accountId))?.asset
     }
+
+    private suspend fun getTagsForTransactionIds(transactions: List<TransactionEntity>): Map<UUID, ImmutableList<Tag>> {
+        val transactionIdList = transactions.map { AssociationId(it.id) }
+        return tagReader.findByAssociatedId(transactionIdList).entries
```

Comment:
We neee to memoize the tags reading in the `tagRepo` using the same pattern as in the `AccountRepository`